### ISSUE TITLE
When SubPage is open, you have to return to the main page to open the menu again.

### ISSUE THREAD

**Issue by saklanmaz** [Patrick386/rail_sub_menu_demo#4]
First of all, sorry for reporting frequently.

For example, I opened SUB-PAGE 2 under Settings page. Then I closed SubMenu. I have to go back to the Settings page to open the SubMenu again. I thought this could be solved like this but I was wondering if you have a better suggestion. SubMenu may be opened automatically when NavigationRail is hovered over. I can do that. Another suggestion, which is the good one; Whichever tab is hovered over in the main menu, its SubMenu is opened. I'm not sure how to do this.

**Comment by Patrick386** [comment]
> First of all, sorry for reporting frequently.
> 
> For example, I opened SUB-PAGE 2 under Settings page. Then I closed SubMenu. I have to go back to the Settings page to open the SubMenu again. I thought this could be solved like this but I was wondering if you have a better suggestion. SubMenu may be opened automatically when NavigationRail is hovered over. I can do that. Another suggestion, which is the good one; Whichever tab is hovered over in the main menu, its SubMenu is opened. I'm not sure how to do this.

If I understood correctly, my first intention was correct. When the user taps on a nail menu, it displays the page set in the _onTabGoScreen method first. If you want to keep showing that page until the user opens another menu, you can do so by making the sub-menu appear when the user hovers over the nail menu. However, the Flutter team has not added the hover feature to the nail menu. That option is already something that I have requested. That's why we have implemented it this way for now. Another way is to remember the state of the open sub-page and pass that state to _onTabGoScreen. This is a practical alternative in the absence of the hover feature.

In conclusion, we have to use it as it is now because the NavigationRail widget does not have the hover feature. Until the Flutter team adds the hover option, we have to continue using it as we do now. If you don't like it, you can customize the NavigationRail. Please refer to this link: https://github.com/flutter/flutter/issues/112270.
